src/components/organisms/PaymentModal/PaymentForm.tsx

```diff
@@ -61,32 +63,63 @@ const PaymentForm: React.FunctionComponent<PropsType> = ({
     }
   }, [event.id, venueId, ticketPrice, user, selectedTab]);
 
-  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+  const handleSubmit = async (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    setIsLoading: (isLoading: boolean) => void,
+    operations: () => Promise<void>
+  ) => {
     if (!user) return;
-    setIsFormBeingSubmitted(true);
+    setIsLoading(true);
     e.preventDefault();
     if (!stripe || !elements) {
       setErrorMessage("Oops something wrong happened");
+      setIsPaymentProceeding(false);
       return;
     }
-    const result = await stripe.confirmCardPayment(clientSecret || "", {
+    await operations();
+    setIsLoading(false);
+  };
+
+  const pay = async () => {
+    const result = await stripe?.confirmCardPayment(clientSecret || "", {
       payment_method: {
-        card: elements.getElement(CardElement) || { token: "" },
+        card: elements?.getElement(CardElement) || { token: "" },
         billing_details: {
-          email: user.email ?? undefined,
+          email: user?.email ?? undefined,
         },
       },
-      receipt_email: user.email ?? undefined,
+      receipt_email: user?.email ?? undefined,
     });
 
-    if (result.error) {
-      setErrorMessage(result.error.message);
+    if (result?.error) {
+      setErrorMessage(result?.error.message);
```

Comment:
if result?.error is true, this means the result object exists and you can remove the `?` in the block

Reply:
.